backend/cli_commands/SyncCommand.py

```diff
@@ -81,10 +74,22 @@ def sync_files(mission_path, mission):
                             mission_id=mission.id,
                             type=typ,
                         )
+                        logging.info(
+                            f"Found file {mcap_path} for mission {mission.name}."
+                        )
                         file.save()
                         logging.info(
                             f"Added new file {mcap_path} for mission {mission.name}."
                         )
+                        for i in metadata:
+                            topic = Topic(
+                                file=file,
+                                name=metadata[i]["name"],
+                                type=metadata[i]["type"],
+                                message_count=metadata[i]["message_count"],
+                                frequency=metadata[i]["frequency"],
+                            )
+                            topic.save()
                     except Exception as e:
                         logging.error(f"Error processing {mcap_path}: {e}")
```

Comment:
```suggestion
                    except Exception as e:
                        logging.error(f"Error processing {mcap_path}: {e}")
                else:
                    file = File.objects.get(file=mcap_path)
                for topic_name,topic_data in topics.items():
                    try:
                        topic = Topic(
                            file=file,
                            name=topic_name,
                            type=topic_data["type"],
                            message_count=topic_data["message_count"],
                            frequency=topic_data["frequency"],
                        )
                        topic.full_clean()
                    except ValidationError as e:
                        logging.error(e)
                    topic.save()
```
This allows adding topics from a file that already is in the database, has a better for loop and runs the validation for a topic.\
But still needs a check if the topic already exists.